src/Codeception/Command/Completion.php

```diff
@@ -0,0 +1,79 @@
+<?php
+namespace Codeception\Command;
+
+use Codeception\Configuration;
+use Stecman\Component\Symfony\Console\BashCompletion\Completion as ConsoleCompletion;
+use Stecman\Component\Symfony\Console\BashCompletion\CompletionCommand;
+use Stecman\Component\Symfony\Console\BashCompletion\CompletionHandler;
+use Stecman\Component\Symfony\Console\BashCompletion\Completion\ShellPathCompletion as ShellPathCompletion;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class Completion extends CompletionCommand
+{
+    protected function configureCompletion(CompletionHandler $handler)
+    {
+        // Can't set for all commands, because it wouldn't work well with generate:suite
+        $suiteCommands = [
+            'run',
+            'config:validate',
+            'console',
+            'dry-run',
+            'generate:cept',
+            'generate:cest',
+            'generate:feature',
+            'generate:phpunit',
+            'generate:scenarios',
+            'generate:stepobject',
+            'generate:test',
+            'gherkin:snippets',
+            'gherkin:steps'
+        ];
+
+        foreach ($suiteCommands as $suiteCommand) {
+            $handler->addHandler(new ConsoleCompletion(
+                $suiteCommand,
+                'suite',
+                ConsoleCompletion::TYPE_ARGUMENT,
+                Configuration::suites()
+            ));
+        }
+
+        $handler->addHandlers([
+            new ShellPathCompletion(
+                ConsoleCompletion::ALL_COMMANDS,
+                'path',
+                ConsoleCompletion::TYPE_ARGUMENT
+            ),
+            new ShellPathCompletion(
+                ConsoleCompletion::ALL_COMMANDS,
+                'test',
+                ConsoleCompletion::TYPE_ARGUMENT
+            ),
+        ]);
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        if ($input->getOption('generate-hook') && $input->getOption('use-vendor-bin')) {
+            global $argv;
+            $argv[0] = 'vendor/bin/' . basename($argv[0]);
```

Comment:
@Naktibalda  There is a difference: although it will autocomplete with just `--program`, it will actually use the codeception version you used to generate the hook, to autocomplete your codeception. I'll give an example scenario:

Say I use autocomplete like so:
`/var/www/myproject/vendor/bin/codecept _completion --generate-hook --program codecept`

In the complete function from the output of `--generate-hook`, it will use the following to autocomplete:
`/var/www/myproject/vendor/bin/codecept _completion`
(see https://github.com/stecman/symfony-console-completion/blob/master/src/HookFactory.php#L148)

With my change, it will complete using the following:
`vendor/bin/codecept _completion`

If you didn't use the `--use-vendor-bin` option, you could run into an issue where it will autocomplete a different project's codeception commands using the original project's codeception.

I could make this behaviour default if that's what you think would be best? But I thought it would be best to leave it open, in case people have different set-ups for their codeception (global codeception for example)

I'll update the docs with more info, including the `eval` option you mentioned.